### PR TITLE
Design doc to md

### PR DESCRIPTION
**Pull request by n3wscott**
Getting a lot of external requests to access the Discovery API doc. Moving it to markdown to make it more accessible.

### CODE REVIEW

File: docs/proposal.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+There was talk of folding Discovery into Eventing directly but I feel it will be
+independently useful for much more than just eventing. It will be a great
+addition for things like Knative CLI, OpenShift UI generators, and GCP UI to
+name a few.
+
+This API will be owned and managed by the Eventing Source WG, it is under the
+charter for sources in a cluster to be discoverable and easily installable.
+
+To get started we need a github.com/knative/discovery repo to be created and we
```
Comment (vaikas) [inline docs/proposal.md:90]: knative-sandbox?

Comment (n3wscott) [reply docs/proposal.md:90]: I just moved the text over...

File: docs/proposal.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Discovery API
+
+| State          | Author(s)     | Last Update               |
+| -------------- | ------------- | ------------------------- |
```
Comment (mattmoor) [inline docs/proposal.md:4]: Format markdown:
```suggestion
| State          | Author(s)     | Last Update                |
| -------------- | ------------- | -------------------------- |
```

File: docs/proposal.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+independently useful for much more than just eventing. It will be a great
+addition for things like Knative CLI, OpenShift UI generators, and GCP UI to
+name a few.
+
+This API will be owned and managed by the Eventing Source WG, it is under the
+charter for sources in a cluster to be discoverable and easily installable.
+
+To get started we need a github.com/knative-sandbox/discovery repo to be
+created and we can start over on API definition, starting with DuckType and
+Manual.
```
Comment (mattmoor) [inline docs/proposal.md:94]: Format markdown:
```suggestion
To get started we need a github.com/knative-sandbox/discovery repo to be created
and we can start over on API definition, starting with DuckType and Manual.
```

File: docs/proposal.md (hunk 2 above)
Comment (mattmoor) [inline docs/proposal.md:4]: Format markdown:
```suggestion
| State          | Author(s)     | Last Update                |
| -------------- | ------------- | -------------------------- |
```

File: docs/proposal.md (hunk 3 above)
Comment (mattmoor) [inline docs/proposal.md:94]: Format markdown:
```suggestion
To get started we need a github.com/knative-sandbox/discovery repo to be created
and we can start over on API definition, starting with DuckType and Manual.
```

**Pull request merged**